src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public static function getSubscribedEvents(): array
+    {
+        return [static::getEventName() => ['__invoke', static::getPriority()]];
+    }
+
+    protected static function getEventName(): string
+    {
+        $method = new \ReflectionMethod(static::class, '__invoke');
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without parameters, you must to define one or implement the "%s::getEventName()" method to return it explicitly.', static::class, $method->getName(), static::class));
+        }
+
+        $parameter = $method->getParameters()[0];
+
+        if (null === $type = $parameter->getType()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without type-hint the "$%s" parameter, you must to type-hint it or implement the "%s::getEventName()" method to return it explicitly.', ContractsEvent::class, $method->getName(), $parameter->getName(), static::class));
+        }
+
+        if ($type->isBuiltin()) {
+            throw new \LogicException(sprintf('The "$%s" parameter defined in "%s::%s()" method must be a class of "%s", "%s" given.', $parameter->getName(), static::class, $method->getName(), ContractsEvent::class, $type->getName()));
```

Comment:
The `Event` class from the contracts is a helper. The error message sounds like we must extend `Event` which is not the case. Virtually any class will do.

Reply:
True ;) fixed, thanks!